gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/deploy/Deployable.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2015-16 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.config.store.deploy;
+
+import java.io.IOException;
+
+import gobblin.config.store.api.ConfigStore;
+
+
+/**
+ * An interface to deploy and rollback {@link ConfigStore}s. {@link ConfigStore}s that implement {@link Deployable} can
+ * be deployed using {@link StoreDeployer}
+ *
+ * @param <D> {@link DeploymentConfig} or its subclasses that has configs to deploy the store
+ * @param <R> {@link RollbackConfig} or its subclasses that has configs to rollback
+ */
+public interface Deployable<D extends DeploymentConfig, R extends RollbackConfig> {
```

Comment:
I think having different config classes is more extensible. We do not yet know the behavior of other ConfigStores. We don't yet know the metadata they may need for deployment and rollback. The rollback may have rollback policies like Rollback if xyz is true etc.